Panic when running pact

Hey there,

I've recently started getting the following error when my pact tests. I can confirm this has been recently introduced, replacing my older version of the package (about 3 weeks old) with the most recent introduces the problem.

```
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x139994f]

goroutine 35 [running]:
testing.tRunner.func1(0xc4201f4680)
	/usr/local/go/src/testing/testing.go:622 +0x29d
panic(0x1408ea0, 0x169f6f0)
	/usr/local/go/src/runtime/panic.go:489 +0x2cf
github.com/pact-foundation/pact-go/dsl.(*PactClient).VerifyProvider(0x0, 0xc42000c14b, 0x15, 0xc4201a5030, 0x1, 0x1, 0x0, 0x0, 0x0, 0x0, ...)
	<hidden>/go/src/github.com/pact-foundation/pact-go/dsl/client.go:118 +0xef
github.com/pact-foundation/pact-go/dsl.(*Pact).VerifyProvider(0xc42024c000, 0xc42000c14b, 0x15, 0xc4201a5030, 0x1, 0x1, 0x0, 0x0, 0x0, 0x0, ...)
	<hidden>/go/src/github.com/pact-foundation/pact-go/dsl/pact.go:219 +0xc2
<hidden>/deploy_test.TestPact(0xc4201f4680)
	<hidden>/pact_verify_test.go:37 +0x442
testing.tRunner(0xc4201f4680, 0x149f800)
	/usr/local/go/src/testing/testing.go:657 +0x96
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:697 +0x2ca
```

--- a/dsl/pact.go
+++ b/dsl/pact.go
@@ -101,6 +101,9 @@ func (p *Pact) Setup(startMockServer bool) *Pact {
 		p.SpecificationVersion = 2
 	}
 
+	client := &PactClient{Port: p.Port, Network: p.Network, Address: p.Host}
+	p.pactClient = client
+
 	if p.Server == nil && startMockServer {
 		args := []string{
 			"--pact-specification-version",
@@ -114,8 +117,6 @@ func (p *Pact) Setup(startMockServer bool) *Pact {
 			"--provider",
 			p.Provider,
 		}
-		client := &PactClient{Port: p.Port, Network: p.Network, Address: p.Host}
-		p.pactClient = client
 		p.Server = client.StartServer(args)
 	}
 

--- a/dsl/pact_test.go
+++ b/dsl/pact_test.go
@@ -182,6 +182,9 @@ func TestPact_Setup(t *testing.T) {
 	if pact.Server != nil {
 		t.Fatalf("Expected server to be nil")
 	}
+	if pact.pactClient == nil {
+		t.Fatalf("Needed to still have a client")
+	}
 }
 
 func TestPact_Teardown(t *testing.T) {

